src/closed.jl

```diff
@@ -0,0 +1,44 @@
+"""
+Defines the ClosedSet type.
+
+A ClosedSet(left, right) is an interval set where both bounds are closed i.e. the range is [left, right]
+"""
+immutable ClosedSet{T}
+	left::T
+	right::T
+end
+
+function ClosedSet(left, right)
+	(I_left, I_right) = promote(left, right)
+	ClosedSet(I_left, I_right)
+end
+
+show(io::IO, I::ClosedSet) = print(io, "[", I.left, ",", I.right, "]")
+
+length(I::ClosedSet) = I.right - I.left + 1
+
+in(v, I::ClosedSet) = I.left <= v <= I.right
+
+isempty(A::ClosedSet) = A.left > A.right
+
+isequal(A::ClosedSet, B::ClosedSet) = (A.left == B.left && A.right == B.right) || isempty(A) && isempty(B)
```

Comment:
This becomes slower with the non-shortcircuiting version. (First benchmark is for the one with `||` and `&&`) - 

```julia
julia> @benchmark isequal(A, B)
BenchmarkTools.Trial: 
  samples:          10000
  evals/sample:     998
  time tolerance:   5.00%
  memory tolerance: 1.00%
  memory estimate:  0.00 bytes
  allocs estimate:  0
  minimum time:     16.00 ns (0.00% GC)
  median time:      16.00 ns (0.00% GC)
  mean time:        16.49 ns (0.00% GC)
  maximum time:     43.00 ns (0.00% GC)
julia> @benchmark isequal(A, B)
BenchmarkTools.Trial: 
  samples:          10000
  evals/sample:     997
  time tolerance:   5.00%
  memory tolerance: 1.00%
  memory estimate:  0.00 bytes
  allocs estimate:  0
  minimum time:     21.00 ns (0.00% GC)
  median time:      21.00 ns (0.00% GC)
  mean time:        21.91 ns (0.00% GC)
  maximum time:     66.00 ns (0.00% GC)
```